### PR TITLE
[CI] Add end-to-end tests

### PR DESCRIPTION
**Pull request by MUzairS15**
Signed-off-by: MUzairS15 <muzair.shaikh810@gmail.com>

**Description**

This PR fixes #
Added end-to-end tests for cilium service mesh installation.
**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [x] Yes, I signed my commits.

### CODE REVIEW

**Comment by MUzairS15** [comment]
Will add the e2e test for the sample application. Stuck in creating a pattern file.

File: .github/workflows/e2etests.yaml [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+    types: [published]
+jobs:
+  SetPatternfile:
+    runs-on: ubuntu-latest
+    outputs:
+      sm_version: ${{ steps.glrt.outputs.release }}
+      adapter_version:  ${{ env.version }}
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@v2
```
Comment (leecalcote) [inline .github/workflows/e2etests.yaml:19]:
```suggestion
        uses: actions/checkout@v3.0.0
```

File: .github/workflows/e2etests.yaml [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        id: glrt
+        run: |
+          NEW_CILIUM_VERSION=$(curl --silent "https://api.github.com/repos/cilium/cilium/releases" | jq ' .[] | select(.tag_name|test("stable")) | ."tag_name"' | sed -n 1p$'\n' | tr -d '"')
+          echo "::set-output name=release::$NEW_CILIUM_VERSION"
+      - name: Change service mesh version in patternfile
+        run: | 
+           yq e -i '.services.cilium.settings.version="${{ steps.glrt.outputs.release }}"' ./.github/install/deploy.yaml 
+           cat ./.github/install/deploy.yaml
+      - name: Uploading file
+        uses: actions/upload-artifact@v2
```
Comment (leecalcote) [inline .github/workflows/e2etests.yaml:37]:
```suggestion
        uses: actions/upload-artifact@v3.0.0
```

File: .github/workflows/e2etests.yaml (hunk 1 above)
Comment (MUzairS15) [reply .github/workflows/e2etests.yaml]: done

File: .github/workflows/e2etests.yaml (hunk 2 above)
Comment (MUzairS15) [reply .github/workflows/e2etests.yaml]: done

File: .github/workflows/e2etests.yaml (hunk 1 above)
Comment (leecalcote) [reply .github/workflows/e2etests.yaml]: thank you.

**Comment by MUzairS15** [comment]
@Zanda256 Can you have a look at why tests are failing? Pods are not ready though they are created.

**Comment by Zanda256** [comment]
> @Zanda256 Can you have a look at why tests are failing? Pods are not ready though they are created.

Let me try and figure it out.@Mohd

**Comment by Zanda256** [comment]
> Will add the e2e test for the sample application. Stuck in creating a pattern file.

I'm also creating one for linkerd. Ping me on slack we can share ideas.

File: .github/workflows/e2etests.yaml [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+        with:
+          name: patternfile
+          path: ./.github/install/deploy.yaml 
+
+  TestCilium:
+    needs: SetPatternfile
+    uses: meshery/meshery/.github/workflows/testadapters.yaml@master
+    with:
+      expected_pods: cilium, cilium-operator, cilium-operator
+      expected_pods_namespaces: kube-system, kube-system, kube-system
```
Comment (Zanda256) [inline .github/workflows/e2etests.yaml:47]: Cilium pods will not come up in Kubernetes-system namespace. Some one correct me if I'm wrong. @gr455 , @Revolyssup

Comment (MUzairS15) [reply .github/workflows/e2etests.yaml:47]: https://github.com/meshery/meshery-cilium/blob/15b4dea1287809a33397f7e2314fe3bc49ad9a94/cilium/install.go#L71

**Comment by warunicorn19** [comment]
@MUzairS15, let's briefly discuss this on the meshery dev call.

File: .github/workflows/e2etests.yaml (hunk 3 above)
Comment (gr455) [reply .github/workflows/e2etests.yaml:47]: > Cilium pods will not come up in Kubernetes-system namespace. Some one correct me if I'm wrong. @gr455 , @Revolyssup

@Zanda256 I think that would depend on what the namespace you give in the pattern file. In this case it's [kube-system](https://github.com/meshery/meshery-cilium/pull/43/files#diff-f1565d9c681714febc630ba3f418bdf416969d152be4d638ff698b02ddde1113R5)

**Comment by leecalcote** [comment]
@Revolyssup ready for your 👀

**Comment by Revolyssup** [comment]
@MUzairS15 cilium operator is trying to kill core-dns pod. This is a know issue. For now, remove `cilium-operator` from assertions for tests to pass and create an issue regarding cilium-operator not starting up.

**Comment by leecalcote** [comment]
@MUzairS15  what do you think?

**Comment by MUzairS15** [comment]
CI failure

```Run eval $(minikube -p minikube docker-env) 
[7](https://github.com/meshery/meshery-cilium/runs/6259034829?check_suite_focus=true#step:4:7)
unable to prepare context: unable to evaluate symlinks in Dockerfile path: lstat /home/runner/work/meshery-cilium/meshery-cilium/install: no such file or directory
[8](https://github.com/meshery/meshery-cilium/runs/6259034829?check_suite_focus=true#step:4:8)
Error: Process completed with exit code 1.

**Comment by Revolyssup** [comment]
> CI failure
> 
> ```
> [7](https://github.com/meshery/meshery-cilium/runs/6259034829?check_suite_focus=true#step:4:7)
> unable to prepare context: unable to evaluate symlinks in Dockerfile path: lstat /home/runner/work/meshery-cilium/meshery-cilium/install: no such file or directory
> [8](https://github.com/meshery/meshery-cilium/runs/6259034829?check_suite_focus=true#step:4:8)
> Error: Process completed with exit code 1.
> ```

Some path changes got into the core workflow. Can you create a PR on the meshery/meshery reverting the change here https://github.com/meshery/meshery/commit/90883a5413c9301efc70d04e4c7f09a4bb88b23b#r72759108

**Comment by MUzairS15** [comment]
> > CI failure
> > ```
> > [7](https://github.com/meshery/meshery-cilium/runs/6259034829?check_suite_focus=true#step:4:7)
> > unable to prepare context: unable to evaluate symlinks in Dockerfile path: lstat /home/runner/work/meshery-cilium/meshery-cilium/install: no such file or directory
> > [8](https://github.com/meshery/meshery-cilium/runs/6259034829?check_suite_focus=true#step:4:8)
> > Error: Process completed with exit code 1.
> > ```
> 
> Some path changes got into the core workflow. Can you create a PR on the meshery/meshery reverting the change here [meshery/meshery@90883a5#r72759108](https://github.com/meshery/meshery/commit/90883a5413c9301efc70d04e4c7f09a4bb88b23b#r72759108) as I have pointed out them. @MUzairS15

Yes sure, will do it.

**Comment by MUzairS15** [comment]
> @MUzairS15 The cilium pod is still not coming up. You think it would need more time than 300 seconds?

300 seconds is enough, because at my fork I checked cilium pod comes up. I will share you screenshot of the log.

**Comment by MUzairS15** [comment]
Screenshot 2022-05-07 at 12 08 43 PM

// @Revolyssup

**Comment by Revolyssup** [comment]
@MUzairS15  some merge conflicts

**Comment by Revolyssup** [comment]
@dependabot rebase

**Pull request merged**